docs: add `brew install hashicorp/tap/consul-k8s` and re-order install and uninstall workflows (#11489)
hashicorp/consul: website/content/docs/k8s/installation/install.mdx

You can install Consul on Kubernetes using the following methods:

1. [Consul K8s CLI install <sup>BETA</sup>.](#consul-k8s-cli-installation)
1. [Helm chart install](#helm-chart-installation)

Refer to the [architecture](/docs/k8s/installation/install#architecture) section to learn more about the general architecture of Consul on Kubernetes.
For a hands-on experience with Consul as a service mesh
for Kubernetes, follow the [Getting Started with Consul service
mesh](https://learn.hashicorp.com/tutorials/consul/service-mesh-deploy?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.

## Consul K8s CLI Installation

> The recommended way to install Consul on Kubernetes for single cluster use cases is via the [Consul K8S CLI](/docs/k8s/k8s-cli). 
> You can install Consul on Kubernetes using the Consul K8s CLI tool after installing the CLI. 
> Ensure that `kubectl` is already configured to authenticate to the Kubernetes cluster using a valid `kubeconfig` file. 
> The Consul K8s CLI is currently available as an beta release and is not recommended for production environments.

Reviewer: ```suggestion
We recommend using the [Consul K8S CLI](/docs/k8s/k8s-cli) to install Consul on Kubernetes for single-cluster deployments. You can install Consul on Kubernetes using the Consul K8s CLI tool after installing the CLI. 

-> **Consul K8s CLI is currently in beta**: The tools is not recommended for production environments.

Before beginning the installation process, verify that `kubectl` is already configured to authenticate to the Kubernetes cluster using a valid `kubeconfig` file. 

The [Homebrew](https://brew.sh) package manager is required to complete the following installation instructions.
```
Just trying to make the language a little cleaner; also mentioned that the CLI tool installation procedure uses brew.

> 1. First, install the HashiCorp tap, a repository of all our Homebrew packages.

Reviewer: ```suggestion
1. Install the HashiCorp `tap`, which is a repository of all Homebrew packages for HashiCorp:
```

   ```shell-session
   brew tap hashicorp/tap
   ```
   

> 1. Next, install the Consul K8s CLI with the `hashicorp/tap/consul` formulae.

Reviewer: ```suggestion
1. Install the Consul K8s CLI with the `hashicorp/tap/consul` formula.
```
To me, it's cleaner to remove words that signal a sequence (i.e., first, next, then, etc.), but I could go either way. Also, is there more than one formula? If so, ignore that suggestion, too.

   ```shell-session
   brew install hashicorp/tap/consul-k8s
   ```
   
1. Issue the `install` subcommand to install Consul on Kubernetes:

   ```shell-session
   consul-k8s install <OPTIONS>
   ```

   Refer to the [Consul K8s CLI reference](/docs/k8s/k8s-cli) for details about all commands and available options.

   If you did not set the `-auto-approve` option to `true`, you will be prompted to proceed with the installation if the pre-install checks pass.

   ```shell-session
   ==> Pre-Install Checks
   ✓ No existing installations found
   ✓ No previous persistent volume claims found
   ✓ No previous secrets found

   ==> Consul Installation Summary
       Installation name: consul
       Namespace: myns
       Overrides:
       connectInject:
         enabled: true
       global:
         name: consul
       server:
         bootstrapExpect: 1
         replicas: 1

       Proceed with installation? (y/n)
   ```

1. Enter `y` to proceed. The pre-install checks may fail if existing `PersistentVolumeClaims` (PVC) are detected. Refer to the [uninstall instructions](/docs/k8s/operations/uninstall#uninstall-consul) for information about removing PVCs.

## Helm Chart Installation

> For multi-cluster installs that involve cross-partition of cross datacenter communication, installing via the Consul on Kubernetes 
> Helm chart is the recommended process. This will install and configure all the necessary components to run Consul. The configuration 
> enables you to run a server cluster, a client cluster, or both.

Reviewer: ```suggestion
We recommend using the Consul Helm chart to install Consul on Kubernetes for multi-cluster installations that involve cross-partition of cross datacenter communication. The Helm chart installs and configures all necessary components to run Consul. The configuration enables you to run a server cluster, a client cluster, or both.
```
Making the language consistent with the previous section.
